### DISCUSSION TITLE
Document VCPKG_ROOT environment variable and make a command to set it

### DISCUSSION THREAD

**Discussion by the-Arioch** [microsoft/vcpkg#24239] [Ideas]
As a newb I was hit by a failure in bootstrapping/building another project: https://github.com/DDoSolitary/yasfw/issues/49

It seems there is a standard (at least de facto standard) `VCPKG_ROOT` env-var, but it is neither mentioned in vcpkg's Quick Start, nor set by vcpkg tool.

Can it be documented (documentation is required to fix the exact name of the variable and to also fix relative path from the variable to the executable (for some tools it can be subfolder like %ROOT%\bin\name.exe)), and mentioned in the readme.md ?

Also, can the bootstrap .bat or vcpkg executable be made to set it? The latter -  as part of `vcpkg integrate install` command or as a separate command like `vcpkg integrate envvar` ? 

At least on Windows platform - for what i remember on Windows NT one needs to update HKCU and/or HKLM registry key, then to broadcast some WM_XXX message to make Explorer re-fetch updated env-vars).

Since `vcpkg integrate install` seems to request UAC elevation anyway, it seems to be a good place to put HKLM env-var adjustment into. If not i'd like to see `vcpkg integrate envvar` and `vcpkg integrate envvar /useronly` commands implemented and mentioned in Quick Start documentation.

**Comment by dg0yt** [comment]
It is not in the quickstart because it isn't needed in regular use cases. But it is documented: 
https://github.com/microsoft/vcpkg/blob/master/docs/users/config-environment.md#vcpkg_root

**Reply by autoantwort** [reply]
This is now https://learn.microsoft.com/en-us/vcpkg/users/config-environment#vcpkg_root

**Comment by the-Arioch** [comment]
`as the root of the vcpkg instance` - is what i was afraid of... Without explicit documenting what is "relative path" between "instance root" and "executable of that instance" it is a guess game. It might be obvious for you that `vcpkg.exe` is expected to be found at `%VCPKG_ROOT%\vcpkg.exe` and not in some subfolder. But it is "internal" knowledge among seasoned people already in-the know (IOW among the people who do not need the documentation).

As of "in regular use cases" it is again a YMMV case. Even if a novice somehow figures he needs to integrate vcpkg into some another toolset - he is lost how exactly to do it. 

A novice one way or another figures he has to plug vcpkg into a pipeline, he googles around and finds https://github.com/microsoft/vcpkg/blob/master/README.md and is hopefully by a single "fineprint" line into https://github.com/microsoft/vcpkg/blob/master/docs/users/integration.md (this link i believe has to be highlighted at least, maybe even put as a separate chapter to be visible from the top of the readme document), and then... hits a wall. Novice user does not immediately use VS IDE, nor does not immediately use CMake. 

I believe there has to be a chapter like "Integrating with other development tools/pipelines" and it has to mention that environment variable as a oficcialyl preferred default way to integrate vcpkg with ANY new 3rd party dev-tools. Make it clear to new users and developers HOW you want them to be finding vcpkg unless any special cases.

* General rule - use THIS env-var in THAT specific way. 
* Special rules for VS IDE
* Special rules for CMake tool
* Special rules for XXXXX tool, etc

**Comment by SiriusTR** [comment]
CMake projects that use CMakePresets.json do typically seem to depend on this environment variable. Without it they don't build (including in VS IDE), so it adds an extra point of friction to have to do this manually.
It's possible that VS eventually plans to set it internally when opening CMake + vcpkg projects; I'm not sure.

**Comment by mcclure** [comment]
I am trying to build a Rust package that requires VCPKG_ROOT to be set. I installed vcpkg today. It installed itself, and the package… somewhere, it never told me where. What value should I set VCPKG_ROOT to to reflect the "default" install location?

I think this env var should be surfaced much higher if major programming languages (Rust?) require it to be set. A comment by @dg0yt links to a doc, but the doc is dead.

**Comment by mcclure** [comment]
Since I asked a question, updating: The VCPKG_ROOT folder appears to refer to (or at least, this is the directory that got Rust to stop giving an error message) the git directory created in the checkout step of the [install instructions](https://vcpkg.io/en/getting-started). The VCPKG_ROOT can be precisely identified by the presence of a file `.vcpkg-root`. None of this was obvious to me and I found it only through blind experimentation.
